src/_guides/libraries/create-library-packages.md

```diff
@@ -153,6 +154,63 @@ it could instead use the `package:` directive
 </aside>
 
 
+## Conditionally importing and exporting library files
+
+If your library supports multiple platforms,
+then you might need to conditionally import or export library files.
+A common use case is a library that supports both web and native platforms.
+
+To conditionally import or export,
+you must check for the presence of `dart:*` libraries.
+Here's an example of conditional export code that
+checks for the presence of `dart:io` and `dart:html`:
+
+<?code-excerpt "create_libraries/lib/hw_mp.dart (export)"?>
+```dart
+export 'src/hw.dart' // Stub implementation
+    if (dart.library.io) 'src/hw_io.dart' // Native|dart:io implementation
+    if (dart.library.html) 'src/hw_html.dart'; // JS|dart:html implementation
+```
+
+Here's how the conditional export code works:
+
+* If this code is running in the Dart VM (JIT or AOT),
+  then export `hw_io.dart`.
+* If this code is running in a browser,
+  then export `hw_html.dart`.
+* Otherwise, export `hw.dart`. **[PENDING: when would this happen?]**
```

Comment:
I think the analyzer will import this during development, so that might be where code completion comes from (check if this is true).

Also if one day a platform that supports neither `dart:html` or `dart:io` then I guess it would be imported there. Though no such platform currently exist.